roles/network_plugin/flannel/tasks/main.yml

```diff
@@ -33,13 +33,21 @@
 - set_fact:
     flannel_mtu: "{{ flannel_mtu_output.stdout }}"
 
-- name: Flannel | Set docker daemon options
-  template:
-    src: docker
-    dest: "/etc/default/docker"
-    owner: root
-    group: root
-    mode: 0644
+- name: Flannel | Set docker daemon options for Debian/CoreOS
+  lineinfile:
+    dest: /etc/default/docker
+    regexp: "# NETWORK_OVERLAY"
+    line: "DOCKER_OPTS=\"$DOCKER_OPTS --bip={{ flannel_subnet }} --mtu={{ flannel_mtu }}\" # NETWORK_OVERLAY"
+  when: ansible_os_family == "Debian" or ansible_os_family == "CoreOS"
+  notify:
+    - restart docker
+
+- name: Flannel | Set docker daemon options for redhat
+  lineinfile:
+    dest: /etc/default/docker
+    regexp: "# NETWORK_OVERLAY"
+    line: "OPTIONS=\"$OPTIONS --bip={{ flannel_subnet }} --mtu={{ flannel_mtu }}\" # NETWORK_OVERLAY"
```

Comment:
Seems like this should be DOCKER_NETWORK_OPTIONS from the previous flannel file.  Also, the OPTIONS case for anything else is missing (or merged into this one)